### PR TITLE
chore(deps): remove dup stylelint depdendencies in webapp/channels

### PR DESCRIPTION
**Pull request by tnir**
#### Summary

In webapp/channels, stylelint deps are both in dependencies and in devDependencies, but it should be only in devDependencies.

This changeset follows up the following PRs:

- #26387
- #26296

and reverts

- #26427
   - automatically generated by [`@unified-ci-app[bot]`](https://github.com/apps/unified-ci-app)
   - cf. #26431 (ported to 9.7 branch)

#### Ticket Link

n/a

#### Screenshots

n/a

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: NOTICE.txt [hunk 1]
```diff
@@ -12173,41 +12173,6 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 
 
----
```
Comment (M-ZubairAhmed) [inline NOTICE.txt:12176]: We will need to keep this as we are not removing the library

**Review by M-ZubairAhmed — CHANGES_REQUESTED** [review]

File: NOTICE.txt (hunk 1 above)
Comment (tnir) [reply NOTICE.txt:12176]: @M-ZubairAhmed No, with this PR, `stylelint` would be only in `devDependencies` as explained in the PR description. See if we do not have `jest` in `NOTICE.txt`, for example, while it (`jest`) is (only) in `devDependencies`: https://github.com/mattermost/mattermost/blob/6d427cf005e4013f230d3f74c51840ce6cc99efd/webapp/channels/package.json#L159

Also you can see as follow:
```
$ cd webapp/channels; npm why stylelint
stylelint@16.2.1 dev
node_modules/stylelint
  stylelint@"^16.2.1" from @stylistic/stylelint-plugin@2.1.0
  node_modules/@stylistic/stylelint-plugin
    dev @stylistic/stylelint-plugin@"2.1.0" from mattermost-webapp@9.3.0
    channels
      mattermost-webapp@9.3.0
      node_modules/mattermost-webapp
        workspace channels from the root project
  dev stylelint@"16.2.1" from mattermost-webapp@9.3.0
  channels
    mattermost-webapp@9.3.0
    node_modules/mattermost-webapp
      workspace channels from the root project
```

Comment (M-ZubairAhmed) [reply NOTICE.txt:12176]: You are absolutely correct

**Review by M-ZubairAhmed — APPROVED** [review]

**Review by hmhealey — APPROVED** [review]
Thanks for catching that! I'm not entirely sure how I moved this by mistake in one of my PRs, but it's good to get that fixed

**Comment by tnir** [comment]
@hmhealey Thanks. Could you check it out https://github.com/mattermost/enterprise/actions/runs/9846765893, which looks like a blocker for merging, as I cannot see it.

**Pull request merged**